include/git2/hook.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_hook_h__
+#define INCLUDE_git_hook_h__
+
+#include "common.h"
+#include "types.h"
+
+/**
+ * @file git2/hook.h
+ * @brief Git hook management routines
+ * @defgroup git_hook Git hook management routines
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * A helper to get the path of the repository's hooks.
+ *
+ * This will obey core.hooksPath.
+ *
+ * @param out_dir A pointer to a git_buf that will be filled with the absolute path.
+ * @param repo The repository to get the hooks location from.
+ * @return 0 on success, or an error code.
+ */
+GIT_EXTERN(int) git_hook_dir(git_buf *out_dir, git_repository *repo);
+
+typedef int (*git_hook_foreach_cb)(const char *hook_name, void *payload);
```

Comment:
`git_hook_foreach` just allows the client to know the list of available hooks in a given repository, which is why you don't get a path here. I don't really see a use case for that — normally, the client just blindly calls `git_hook_execute()`, which will cause the registered executor to be called with the required parameters if the hook exists and is valid. In other words, the client doesn't need to know whether the hook is available or not.

My thought would be to replace all that with a `git_hook_exists(git_repository *repo, const char *hook_name)`, but I have no idea if it's worth it...